neqo-transport/tests/server.rs

```diff
@@ -160,20 +161,20 @@ fn duplicate_initial_new_path() {
         &initial[..],
     );
 
-    // The server should respond to both as these came from different addresses.
-    let dgram = server.process(Some(&other), now()).dgram();
-    assert!(dgram.is_some());
-
     let server_initial = server.process(Some(&initial), now()).dgram();
     assert!(server_initial.is_some());
 
-    assert_eq!(server.active_connections().len(), 2);
-    complete_connection(&mut client, &server, server_initial);
+    // The server should ignore a packet with the same destination connection ID.
+    let dgram = server.process(Some(&other), now()).dgram();
+    assert!(dgram.is_none());
+
+    assert_eq!(server.active_connections().len(), 1);
+    complete_connection(&mut client, &mut server, server_initial);
 }
```

Comment:
I believe a server SHOULD  ignore the second packet, given that it carries the same destination connection ID and packet number. In other words, I don't think the previous assertion of the test is correct.

The previous assertion (i.e. _should not ignore_) now fails with this pull request, as connections are no longer lazily indexed through the `ConnectionIdGenerator` wrapper, but instead looked-up ad-hoc on each packet.

Calling this out explicitly to make sure I am not missing something.

Reply:
Yeah, I don't understand either why the server should respond to both. https://datatracker.ietf.org/doc/html/rfc9000#section-21.2 says
> Addresses cannot change during the handshake, so endpoints can discard packets that are received on a different network path.

@martinthomson may have an opinion.

Reply:
Marking this as resolved. We can revisit this in case @martinthomson thinks the old behavior was correct.